include/nbl/builtin/hlsl/workgroup/fft.hlsl

```diff
@@ -157,28 +159,101 @@ struct FFT<2,true, Scalar, device_capabilities>
 
 // ---------------------------- Below pending --------------------------------------------------
 
-/*
-
-// then define 4,8,16 in terms of calling the FFT<2> and doing the special radix steps before/after
-template<uint16_t K, bool Inverse, class device_capabilities>
-struct FFT
+// Forward FFT
+template<uint32_t K, typename Scalar, class device_capabilities>
+struct FFT<K, false, Scalar, device_capabilities>
 {
-    template<typename Accessor, typename ShaderMemoryAccessor>
-    static void __call(NBL_REF_ARG(Accessor) accessor, NBL_REF_ARG(ShaderMemoryAccessor) sharedmemAccessor)
+    template<typename Accessor, typename SharedMemoryAccessor>
+    static enable_if_t< (mpl::is_pot_v<K> && K > 2), void > __call(NBL_REF_ARG(Accessor) accessor, NBL_REF_ARG(SharedMemoryAccessor) sharedmemAccessor)
     {
-        if (!Inverse)
+        static const uint32_t virtualThreadCount = K >> 1;
+        static const uint16_t passes = mpl::log2<K>::value - 1;
+        uint32_t stride = K >> 1;
+        //[unroll(passes)]
+        for (uint16_t pass = 0; pass < passes; pass++)
```

Comment:
alternative way to formulate is also `for (uint32_t stride = K<<(WORKGROUP_SIZE_LOG2-1); stride>WORKGROUP_SIZE; stride>>=1)`

Reply:
new math is good